Add item requests and gateway

PAZderev opened the pull request.

PAZderev committed:
feat (ItemRequest) : Добавление новой функциональности - запросы на предметы
 - Добавлены классы слоя приложения, сервиса, репозитория для ItemRequest
 - Добавлено новое поле в Item
 - Учтено новое поле в Item в методах сервиса ItemServiceImpl
 - Создана новая таблица в БД для ItemRequest

PAZderev committed:
feat : Разделение проекта на микросервисы
 - Проект разделен на два модуля - gateway и server

PAZderev committed:
test : Добавление юнит, интеграционных, json тестов для всего проекта

db-exp commented on gateway/src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -2,12 +2,14 @@
+@AllArgsConstructor
 public class ItemDto {
     private Long id;
     @NotBlank(message = "name must be not blank")
```
+Size

db-exp commented on server/pom.xml:
```diff
@@ -0,0 +1,81 @@
+            <scope>runtime</scope>
+        </dependency>
+
+        <dependency>
```
для сервера  validation не нужен

PAZderev committed:
fix : Правки по комментариям
 - Добавил валидацию размера в ItemDto
 - Убрал лишние зависимости в модуле server
